diff --git a/node_modules/@op-engineering/op-sqlite/android/cpp-adapter.cpp b/node_modules/@op-engineering/op-sqlite/android/cpp-adapter.cpp
index 5912d7b..495880e 100644
--- a/node_modules/@op-engineering/op-sqlite/android/cpp-adapter.cpp
+++ b/node_modules/@op-engineering/op-sqlite/android/cpp-adapter.cpp
@@ -19,8 +19,8 @@ struct OPSQLiteBridge : jni::JavaClass<OPSQLiteBridge> {
   static void registerNatives() {
     javaClassStatic()->registerNatives(
         {makeNativeMethod("installNativeJsi", OPSQLiteBridge::installNativeJsi),
-         makeNativeMethod("clearStateNativeJsi",
-                          OPSQLiteBridge::clearStateNativeJsi)});
+         makeNativeMethod("clearStateNativeJsi", OPSQLiteBridge::clearStateNativeJsi),
+    	 makeNativeMethod("deleteAllDBsJsi", OPSQLiteBridge::deleteAllDBsJsi)});
   }
 
 private:
@@ -39,6 +39,10 @@ private:
   static void clearStateNativeJsi(jni::alias_ref<jni::JObject> thiz) {
     opsqlite::invalidate();
   }
+
+  static bool deleteAllDBsJsi(jni::alias_ref<jni::JObject> thiz) {
+  	return opsqlite::deleteAllDbs();
+  }
 };
 
 JNIEXPORT jint JNI_OnLoad(JavaVM *vm, void *) {
diff --git a/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteBridge.kt b/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteBridge.kt
index 44f86df..8c003db 100644
--- a/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteBridge.kt
+++ b/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteBridge.kt
@@ -12,6 +12,7 @@ class OPSQLiteBridge {
         docPath: String
     )
     private external fun clearStateNativeJsi()
+    private external fun deleteAllDBsJsi(): Boolean
 
     fun install(context: ReactContext) {
         val jsContextPointer = context.javaScriptContextHolder!!.get()
@@ -31,6 +32,10 @@ class OPSQLiteBridge {
         clearStateNativeJsi()
     }
 
+    fun deleteAllDBs(): Boolean {
+        return deleteAllDBsJsi()
+    }
+
     companion object {
         val instance = OPSQLiteBridge()
     }
diff --git a/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteModule.kt b/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteModule.kt
index 8d341b7..33d3d7d 100644
--- a/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteModule.kt
+++ b/node_modules/@op-engineering/op-sqlite/android/src/main/java/com/op/sqlite/OPSQLiteModule.kt
@@ -12,7 +12,7 @@ import java.io.InputStream
 import java.io.OutputStream
 import com.facebook.react.util.RNLog;
 
-internal class OPSQLiteModule(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {
+class OPSQLiteModule(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {
     override fun getName(): String {
         return NAME
     }
@@ -54,6 +54,18 @@ internal class OPSQLiteModule(context: ReactApplicationContext?) : ReactContextB
         throw Exception("Do not call getDylibPath on Android")
     }
 
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    fun closeAllConnections(): Boolean {
+        OPSQLiteBridge.instance.invalidate()
+        return true
+    }
+
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    fun deleteAllDBs(): Boolean {
+        return OPSQLiteBridge.instance.deleteAllDBs();
+    }
+
+
     @ReactMethod
     fun moveAssetsDatabase(args: ReadableMap, promise: Promise) {
         val filename = args.getString("filename")!!
diff --git a/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.cpp b/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.cpp
index 9682941..c0b9c21 100644
--- a/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.cpp
+++ b/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.cpp
@@ -682,6 +682,10 @@ void DBHostObject::invalidate() {
 #endif
 }
 
+void DBHostObject::drop() {
+    opsqlite_remove(db, db_name, std::string(base_path));
+}
+
 DBHostObject::~DBHostObject() { invalidate(); }
 
 } // namespace opsqlite
diff --git a/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.h b/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.h
index 002b8c9..f3a2e1f 100644
--- a/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.h
+++ b/node_modules/@op-engineering/op-sqlite/cpp/DBHostObject.h
@@ -68,6 +68,7 @@ public:
   void on_commit();
   void on_rollback();
   void invalidate();
+  void drop();
   ~DBHostObject() override;
 
 private:
diff --git a/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.cpp b/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.cpp
index e33579a..791fa70 100644
--- a/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.cpp
+++ b/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.cpp
@@ -42,6 +42,13 @@ void invalidate() {
   dbs.clear();
 }
 
+bool deleteAllDbs() {
+    for(const auto &db : dbs) {
+        db->drop();
+    }
+    return true;
+}
+
 void install(jsi::Runtime &rt,
              const std::shared_ptr<react::CallInvoker> &_invoker,
              const char *base_path, const char *crsqlite_path,
diff --git a/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.hpp b/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.hpp
index 91511ab..cc73dfe 100644
--- a/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.hpp
+++ b/node_modules/@op-engineering/op-sqlite/cpp/OPSqlite.hpp
@@ -14,6 +14,7 @@ void install(jsi::Runtime &rt,
              const char *base_path, const char *crsqlite_path,
              const char *sqlite_vec_path);
 void invalidate();
+bool deleteAllDbs();
 void expoUpdatesWorkaround(const char *base_path);
 
 } // namespace opsqlite
